ui/app/models/transform.js

```diff
@@ -40,81 +39,106 @@ const TWEAK_SOURCE = [
   },
 ];
 
-export default Model.extend({
-  name: attr('string', {
-    // CBS TODO: make this required for making a transformation
-    label: 'Name',
+// TODO: add modelValidations, requires restructuring the error message flow into the components.
+export default class TransformModel extends Model {
+  @attr('string', {
+    readOnly: true,
+  })
+  backend;
+
+  @attr('string', {
     readOnly: true,
     subText: 'The name for your transformation. This cannot be edited later.',
-  }),
-  type: attr('string', {
+  })
+  name;
+
+  @attr('string', {
     defaultValue: 'fpe',
     label: 'Type',
     possibleValues: TYPES,
     subText:
       'Vault provides two types of transformations: Format Preserving Encryption (FPE) is reversible, while Masking is not. This cannot be edited later.',
-  }),
-  tweak_source: attr('string', {
+  })
+  type;
+
+  @attr('string', {
     defaultValue: 'supplied',
     label: 'Tweak source',
     possibleValues: TWEAK_SOURCE,
     subText: `A tweak value is used when performing FPE transformations. This can be supplied, generated, or internal.`, // CBS TODO: I do not include the link here.  Need to figure out the best way to approach this.
-  }),
-  masking_character: attr('string', {
+  })
+  tweak_source;
+
+  @attr('string', {
     characterLimit: 1,
     defaultValue: '*',
     label: 'Masking character',
     subText: 'Specify which character you’d like to mask your data.',
-  }),
-  template: attr('array', {
+  })
+  masking_character;
+
+  // TODO: add presence required validation for template
```

Comment:
This is the only real change on the model, everything else is due to glimmerizing